core/Pos/Crypto/Scrypt.hs

```diff
@@ -0,0 +1,66 @@
+-- | Wrapper over scrypt library
+
+module Pos.Crypto.Scrypt
+    ( S.ScryptParams
+    , S.Salt (..)
+    , S.Pass (..)
+    , S.EncryptedPass (..)
+
+    , ScryptParamsBuilder (..)
+    , mkScryptParams
+
+    , mkSalt
+    , genSalt
+    , encryptPassIO
+    , encryptPassWithSalt
+    , verifyPass
+    ) where
+
+import           Universum
+
+import qualified Crypto.Scrypt     as S
+import           Data.Default      (Default (..))
+
+import           Pos.Binary.Class  (Bi, encodeStrict)
+import           Pos.Crypto.Random (secureRandomBS)
+
+-- | This corresponds to 'ScryptParams' datatype.
+-- These parameters influence on resulting hash length, memory and time
+-- consumption. See documentation for exact details.
+data ScryptParamsBuilder = ScryptParamsBuilder
+    { spLogN    :: Word
+    , spR       :: Word
+    , spP       :: Word
```

Comment:
You're right, it became unclear. 
In your opinion, if I attach this link, will it be enough? https://hackage.haskell.org/package/scrypt-0.5.0/docs/Crypto-Scrypt.html